proto/token.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+import "permissions.proto";
+
+option go_package = "github.com/momentohq/client-sdk-go;client_sdk_go";
+option java_multiple_files = true;
+option java_package = "momento.auth";
+
+package token;
+
+service Token {
+  rpc GenerateTempAuthToken (_GenerateTempAuthTokenRequest) returns (stream _GenerateTempAuthTokenResponse) {}
+}
+
+message _GenerateTempAuthTokenRequest {
+    // generate a token that has an expiry
```

Comment:
GenerateAuthToken used to take the source token used for the call as an input parameter `auth_token` - I don't remember why. Does this api use the token from the Authorization header?

Reply:
it was because some of the APIs required a session token and others required an API token, and it seemed more clear to make them input parameters so that we weren't overloading the meaning of the header.

Unless we have a good reason to change things up I think I'd vote for consistency again and just say that all Auth APIs accept their required auth tokens as inputs. It's more explicit and less magical.

Reply:
Ah, this is a good point, the auth clients for our sdk's don't send the token over the auth headers. I was sort of making the assumption that we would be making a new client for this, would want to get @cprice404 thoughts on that.